### ISSUE TITLE
`[ttyc 13:39:33] unable to connect or authenticate to server: failed to WebSocket dial: expected handshake response status code 101 but got 401`

### ISSUE THREAD

**Issue by rihaq** [Depau/ttyc#2]

**Comment by Depau** [comment]
I can't reproduce the issue. You need to tell me more about your setup.

**Comment by rihaq** [comment]
> I can't reproduce the issue. You need to tell me more about your setup.

Using caddy. ttyd at https://domain.tld/portal/ttyd
Ttyd version: 1.7.1

**Comment by Depau** [comment]
Yes but what's your caddyfile? How are you starting `ttyd`? Please give me a step-by-step guide to reproduce and test your setup, otherwise I can't help you. It works on my machine.

**Comment by rihaq** [comment]
> Yes but what's your caddyfile? How are you starting `ttyd`? Please give me a step-by-step guide to reproduce and test your setup, otherwise I can't help you. It works on my machine.

Caddyfile.
```
handle portal/ttyd* {
        redir portal/ttyd portal/ttyd/
        uri strip_prefix portal/ttyd       
        basicauth {              
                rihaq <hashed_pass>
        }     
        reverse_proxy * localhost:61803
}
```

Then I run kaddy like this
`caddy run --config Caddyfile --adapter caddyfile`

And run ttyd like this
`ttyd -i 127.0.0.1 -p 61803 -P 3 -t fontSize=17 -t titleFixed=ttyd -t 'theme={"background": "black"}' bash --rcfile /home/rihaq/bashrc`

**Comment by Depau** [comment]
Hi, I'm sorry but I'm not going to fix it, it would need a refactor that I don't have time to do.
I'll leave it open in case anyone wants to fix it as an improvement.

The issue is that `ttyd` username/password authentication only uses the proper basic auth to get a token, then it uses the token to continue the session. But if you use caddy to do the authentication it will require the authentication on all paths, which `ttyc` does not handle.

Possible workarounds:

1. `ttyd -c user:pass`
2. If you want multiple users you can still use `ttyd -c user:pass`, configure caddy to require basic auth on `/` and `/token`, then configure the reverse proxy to send the correct `Authorization: Basic xxx` header to `ttyd`.
